### PR TITLE
DD4hep: SimTracker/TrackerMaterialAnalysis Migration MaterialAccounting

### PR DESCRIPTION
**Pull request by vargasa**
#### PR description:

Progress on `dd4hep` migration on `SimTracker/TrackerMaterialAnalysis`

#### PR validation:
DD
```
$ cmsRun SimTracker/TrackerMaterialAnalysis/test/listGroups.py
```
DD4hep
```
$ cmsRun SimTracker/TrackerMaterialAnalysis/test/listGroups_dd4hep.py
```

### CODE REVIEW

**Comment by vargasa** [comment]
FYI: @cvuosalo @ianna

**Comment by cvuosalo** [comment]
@cmsbuild please test

File: SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+  for (const auto j : fv.specpars()) {
+    for (const auto k : j->paths) {
+      if (firstChild) {
+        std::vector<std::vector<cms::Node*>> children = fv.children(k);
+        for (auto const& path : children) {
+          cms::Translation trans = fv.translation(path);
+          GlobalPoint gp = GlobalPoint(trans.x(), trans.y(), trans.z());
+          m_elements.push_back(gp);
+          std::cout << "Adding element at (r,z) " << gp.perp() << "," << gp.z() << std::endl;
```
Comment (cvuosalo) [inline SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc:43]: Shouldn't all these `cout` statements be changed to `LogVerbatim`? Is this program only for testing?

**Comment by civanch** [comment]
+1
I would guess it is test code, so "cout" is allowed

File: SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc (hunk 1 above)
Comment (vargasa) [reply SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc:43]: @cvuosalo This specific part has not changed from the existing code. Due to time constraints I'm focusing on having it working and we can address these details in future iterations

File: SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    }
+  }
+
+  for (unsigned int i = 0; i < m_elements.size(); ++i) {
+    m_boundingbox.grow(m_elements[i].perp(), m_elements[i].z());
+  }
+
+  m_boundingbox.grow(s_tolerance);
+
+  std::cout << "Final BBox r_range: " << m_boundingbox.range_r().first << ", " << m_boundingbox.range_r().second
```
Comment (silviodonato) [inline SimTracker/TrackerMaterialAnalysis/plugins/dd4hep/DD4hep_MaterialAccountingGroup.cc:55]: also here

**Comment by boudoul** [comment]
Hi @silviodonato and @vargasa , may I kindly suggest to one of you to create an issue opened until the cout is fixed, so that we are not forgetting ... thanks

**Comment by vargasa** [comment]
@boudoul @silviodonato thanks, I'll take care of it

**Pull request merged**